src/Api/SecretsManager/Controllers/SecretsController.cs

```diff
@@ -43,37 +55,131 @@ public async Task<SecretResponseModel> GetSecretAsync([FromRoute] Guid id)
         {
             throw new NotFoundException();
         }
+        
+        if(!await userHasReadAccessToProject(secret))
+        {
+            throw new UnauthorizedAccessException();
+        }
+
         return new SecretResponseModel(secret);
     }
 
-    [HttpGet("projects/{projectId}/secrets")]
-    public async Task<SecretWithProjectsListResponseModel> GetSecretsByProjectAsync([FromRoute] Guid projectId)
+    [HttpGet("projects/{projectId}/{organizationId}/secrets")]
+    public async Task<SecretWithProjectsListResponseModel> GetSecretsByProjectAsync([FromRoute] Guid projectId, [FromRoute] Guid organizationId)
     {
-        var secrets = await _secretRepository.GetManyByProjectIdAsync(projectId);
+        var userId = _userService.GetProperUserId(User).Value;
+        var orgAdmin = await _currentContext.OrganizationAdmin(organizationId);
+        var accessClient = AccessClientHelper.ToAccessClient(_currentContext.ClientType, orgAdmin);
+
+        var secrets = await _secretRepository.GetManyByProjectIdAsync(projectId, userId, accessClient);
+        
+        if(secrets != null){
+            if(!await userHasReadAccessToProject(projectId, organizationId))
```

Comment:
Don't need access checks here now that we are filtering in GetManyByProjectIDAsync()